### PR TITLE
Use custom pretty printer in logs

### PR DESCRIPTION
**Pull request by pawelprazak**

### CODE REVIEW

File: core/src/main/scala/besom/internal/ResourceDecoder.scala [hunk 1]
```diff
@@ -39,12 +41,14 @@ object ResourceDecoder:
               .decode(value, propertyLabel)
               .map(_.withDependency(resource))
               .tapBoth(
-                err => log.debug(s"failed to extract custom property '$propertyName' from '$value': '$err'"),
-                value => log.debug(s"extracted custom property '$propertyName' from '$value'")
+                err => log.debug(s"failed to extract custom property '$propertyName' from '${printer.render(value)}': '$err'"),
+                _ => log.debug(s"extracted custom property '$propertyName' from '${printer.render(value)}'")
```
Comment (lbialy) [inline core/src/main/scala/besom/internal/ResourceDecoder.scala:45]: Wait, this does something different than before, now it prints the value as it was before decoding. Before it printed the result of decoding, didn't it?

File: core/src/main/scala/besom/internal/ResourceOps.scala [hunk 2]
```diff
@@ -487,7 +487,7 @@ class ResourceOps(using ctx: Context, mdc: BesomMDC[Label]):
             ,
             version = inputs.options.version.getOrElse(""), // protobuf expects empty string and not null
             ignoreChanges = inputs.options.ignoreChanges,
-            acceptSecrets = true, // TODO doing like java does it
+            acceptSecrets = true, // our implementation supports secrets form day one
```
Comment (lbialy) [inline core/src/main/scala/besom/internal/ResourceOps.scala:490]: form -> from

File: core/src/main/scala/besom/util/printer.scala [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+  override def treeify(x: Any): Tree =
+    val result = super.treeify(x)
+    result match
+      case Tree.Apply(p, body: Iterator[Tree]) =>
+        Tree.Apply(
+          p,
+          body
+            .filterNot {
+              case Tree.KeyValue("unknownFields", _) => true
```
Comment (lbialy) [inline core/src/main/scala/besom/util/printer.scala:20]: Great!

**Review by lbialy — CHANGES_REQUESTED** [review]
Smol nitpicks, one dubious change. Overall score: veri nice.

File: core/src/main/scala/besom/internal/ResourceDecoder.scala (hunk 1 above)
Comment (pawelprazak) [reply core/src/main/scala/besom/internal/ResourceDecoder.scala:45]: This error message is not correct, I have a correction in another branch.

The `value` in the `tapBoth` was shadowing the `value` in `map` above.
The decoded value will be full of outputs (the ones you've complained about).

Comment (pawelprazak) [reply core/src/main/scala/besom/internal/ResourceDecoder.scala]: I've reverted the debug message to display the resulting output data, but changed the variable name to get rid of shadowing.

I've changed the printer to turn `besom.internal.Output@3b751a0c` into `Output(?)` in logs.

**Review by lbialy — APPROVED** [review]

**Pull request merged**